pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -772,6 +779,146 @@ func GatherMachineSet(i *Gatherer) func() ([]record.Record, []error) {
 	}
 }
 
+// GatherInstallPlans collects Top x InstallPlans.
+// Because InstallPlans have unique generated names, it groups them by namespace and the "template"
+// for name generation from field generateName.
+// It also collects Total number of all installplans and all non-unique installplans.
+//
+// The Kubernetes api https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/certificates/v1beta1/certificatesigningrequest.go#L78
+// Response see https://docs.openshift.com/container-platform/4.3/rest_api/index.html#certificatesigningrequestlist-v1beta1certificates
+//
+// Location in archive: config/certificatesigningrequests/
+func GatherInstallPlans(i *Gatherer) func() ([]record.Record, []error) {
+	return func() ([]record.Record, []error) {
+		var plansBatchLimit int64 = 500
+
+		cont := ""
+		recs := map[string]collectedPlan{}
+		tc := 0
+
+		// oc get installplans -n=openshift-operators -v=6
+		opResource := schema.GroupVersionResource{Group: "operators.coreos.com", Version: "v1alpha1", Resource: "installplans"}
+		resInterface := i.dynamicClient.Resource(opResource).Namespace("openshift-operators")
+		for {
+			u, err := resInterface.List(i.ctx, metav1.ListOptions{Limit: plansBatchLimit, Continue: cont})
+			if errors.IsNotFound(err) {
+				return nil, nil
+			}
+			if err != nil {
+				return nil, []error{err}
+			}
+
+			jsonMap := u.UnstructuredContent()
+			var items []interface{}
+			err = failEarly(
+				func() error { return parseJSONQuery(jsonMap, "metadata.continue?", &cont) },
+				func() error { return parseJSONQuery(jsonMap, "items", &items) },
+			)
+			if err != nil {
+				return nil, []error{err}
+			}
+
+			tc += len(items)
+			for _, item := range items {
+				// Get common prefix
+				csv := "[NONE]"
+				var clusterServiceVersionNames []interface{}
+				var ns, genName string
+				var itemMap map[string]interface{}
+				var ok bool
+				if itemMap, ok = item.(map[string]interface{}); !ok {
+					if err != nil {
+						return nil, []error{fmt.Errorf("cannot cast item to map %v", item)}
+					}
+				}
+
+				err = failEarly(
+					func() error {
+						return parseJSONQuery(itemMap, "spec.clusterServiceVersionNames", &clusterServiceVersionNames)
+					},
+					func() error { return parseJSONQuery(itemMap, "metadata.namespace", &ns) },
+					func() error { return parseJSONQuery(itemMap, "metadata.generateName", &genName) },
+				)
+				if err != nil {
+					return nil, []error{err}
+				}
+				if len(clusterServiceVersionNames) > 0 {
+					// ignoring non string
+					csv, _ = clusterServiceVersionNames[0].(string)
+				}
+
+				key := fmt.Sprintf("%s.%s.%s", ns, genName, csv)
+				m, ok := recs[key]
+				if !ok {
+					recs[key] = collectedPlan{Namespace: ns, Name: genName, CSV: csv, Count: 1}
+				} else {
+					m.Count++
+				}
```

Comment:
interesting PR overall :)

optional:
My personal preference would be to extract processing one item into a function with a descriptive name for better readability.
(if it makes sense for you in this case)